x-pack/plugins/endpoint/server/endpoint_app_context_services.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { IndexPatternRetriever } from './index_pattern';
+import { AgentService } from '../../ingest_manager/common/types';
+
+/**
+ * A singleton that holds shared services that are initialized during the start up phase
+ * of the plugin lifecycle. And stop during the stop phase, if needed.
+ */
+class EndpointAppContextService {
+  private indexPatternRetriever: IndexPatternRetriever | undefined;
+  private agentService: AgentService | undefined;
+
+  public start(dependencies: {
+    indexPatternRetriever: IndexPatternRetriever;
+    agentService: AgentService;
+  }) {
+    this.indexPatternRetriever = dependencies.indexPatternRetriever;
+    this.agentService = dependencies.agentService;
+  }
+
+  public stop() {}
```

Comment:
:grey_question: I _think_ this is OK, but just checking - are there cases where you'd ever want to stop and restart this service? If you did, would you want to enforce starting it with a new `indexPatternRetriever` / `agentService`